metalearn/metafeatures/perceptron_metafeatures.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+
+from sklearn.linear_model import Perceptron
+
+from metalearn.metafeatures.common_operations import profile_distribution
+
+
+def get_fitted_perceptron(X, Y, seed, n_classes, frac='all'):
+    class_fraction = n_classes/len(X)
+
+    if frac == 'all':
+        frac = class_fraction
+    elif frac == 'tenth':
+        frac = min(max(0.9, class_fraction), 1 - class_fraction)
+    elif frac == 'half':
+        frac = min(max(0.5, class_fraction), 1 - class_fraction)
+    elif frac == 'sqrt':
+        frac = min(max(1 - (np.sqrt(len(X)) / len(X)), class_fraction), 1 - class_fraction)
+
+    cls = Perceptron(random_state=seed, validation_fraction=frac, early_stopping=True, max_iter=1000, tol=1e-3)
+
+    if n_classes == 1:
```

Comment:
@macetheace96, do you think it would help to add a little comment here explaining what is being done, since it's not straightforward? E.g.:
```suggestion
    if n_classes == 1:
    # The `.fit` method cannot handle a dataset with 1 class, so we
    # give the model the theoretical weights it would learn on a single
    # class classification problem: zero weights (so as to predict class 0
    # every time). 
```